source/adios2/CMakeLists.txt

```diff
@@ -71,15 +71,16 @@ endif()
   
 if(ADIOS2_HAVE_BZip2)
   find_package(BZip2 REQUIRED)
-  target_sources(adios2 PRIVATE transform/compression/BZip2.cpp)
-  target_link_libraries(adios2 PRIVATE BZip2::BZip2)
+  target_sources(adios2 PRIVATE transform/compress/CompressBZip2.cpp)
+  target_include_directories(adios2 PUBLIC ${BZIP2_INCLUDE_DIR})
+  target_link_libraries(adios2 PUBLIC ${BZIP2_LIBRARIES})
 endif()
 
 if(ADIOS2_HAVE_ZFP)
   find_package(ZFP REQUIRED)
-  message("ADIOS ZFP support not yet implemented")
-#  target_sources(adios2 PRIVATE transform/compression/ZFP.cpp)
-#  target_link_libraries(adios2 PRIVATE zfp::zfp)
+  target_sources(adios2 PRIVATE transform/compress/CompressZfp.cpp)
+  target_include_directories(adios2 PUBLIC ${ZFP_INCLUDE_DIRS})
+  target_link_libraries(adios2 PUBLIC zfp::zfp)
```

Comment:
Drop the include dirs, they should already be provided by the imported target.  zfp should also remain a private dependency.